robot-server/tests/sessions/test_engine_store.py

```diff
@@ -0,0 +1,47 @@
+"""Integration tests for the EngineStore interface."""
+import pytest
+from mock import MagicMock
+
+from opentrons.protocol_engine import ProtocolEngine
+from robot_server.sessions.engine_store import (
+    EngineStore,
+    EngineConflictError,
+    EngineMissingError,
+)
+
+
+@pytest.fixture
+def subject() -> EngineStore:
+    """Get a EngineStore test subject."""
+    return EngineStore(hardware_api=MagicMock())
+
+
+async def test_create_engine(subject: EngineStore) -> None:
+    """It should create an engine."""
+    result = await subject.create()
+
+    assert isinstance(result, ProtocolEngine)
+    assert isinstance(subject.engine, ProtocolEngine)
+
+
+async def test_raise_if_engine_already_exists(subject: EngineStore) -> None:
+    """It should not create more than one engine / runner pair."""
+    await subject.create()
+
+    with pytest.raises(EngineConflictError):
+        await subject.create()
+
+
+def test_raise_if_engine_does_not_exist(subject: EngineStore) -> None:
+    """It should raise if no engine exists when requested."""
+    with pytest.raises(EngineMissingError):
+        subject.engine
```

Comment:
Should we also test here that removal fails if there's nothing to remove?

```suggestion
def test_raise_if_engine_does_not_exist(subject: EngineStore) -> None:
    """It should raise if no engine exists when requested."""
    with pytest.raises(EngineMissingError):
        subject.engine

    with pytest.raises(EngineMissingError):
        subject.remove()
```

(Not sure if we should additionally duplicate this new `with pytest.raises` block in `test_remove_engine()`, too...)

Reply:
Going to say no for now due to lack of need; if something needs the engine gone and the engine is already gone, then that's a good state.

Considering renaming `remove` to reflect this behavior, or at least putting in the docstring, and we need to add a test for this no-op behavior